python/sdk/merlin/model.py

```diff
@@ -1029,82 +1029,67 @@ def deploy(self, environment_name: str = None,
         :param protocol: protocol to be used for deploying the model (default: HTTP_JSON)
         :return: VersionEndpoint object
         """
+        current_endpoint = self.endpoint
+        env_list = self._get_env_list()
```

Comment:
This call reduces all the redundant calls to the ListEnvironments API to just 1 call at the start of the entire method.